Formula/mbedtls.rb

```diff
@@ -1,8 +1,8 @@
 class Mbedtls < Formula
   desc "Cryptographic & SSL/TLS library"
   homepage "https://tls.mbed.org/"
-  url "https://github.com/Mbed-TLS/mbedtls/archive/mbedtls-3.1.0.tar.gz"
-  sha256 "64d01a3b22b91cf3a25630257f268f11bc7bfa37981ae6d397802dd4ccec4690"
+  url "https://github.com/Mbed-TLS/mbedtls/archive/mbedtls-3.2.1.tar.gz"
+  sha256 "5850089672560eeaca03dc36678ee8573bb48ef6e38c94f5ce349af60c16da33"
   license "Apache-2.0"
   head "https://github.com/Mbed-TLS/mbedtls.git", branch: "development"
```

Comment:
Probably should figure out how we want to handle HEAD as it doesn't work in current formula. Either need to drop HEAD support or update the formula to properly build development branch.

Reply:
No idea what's the best course of action here. And indeed I watched too quickly as I've seen the file added in the master… but removed right after: https://github.com/Mbed-TLS/mbedtls/commit/af4b90db3fbc06874747c7c967d2a8a21f6ae587

Reply:
The upstream only adds generated files for short time to do an official release. It is generally expected that generated files will not be in git repo, so `GEN_FILES=OFF` would have broken HEAD build (since Jan 2022 -- https://github.com/Homebrew/homebrew-core/commit/75b328376e7d1418bc8d403096e774b228e98742).

The main troublesome thing with HEAD builds is if upstream adds any other dependencies beyond Jinja. Simplest, but not necessarily ideal, way would be to use upstream's requirements file and just let pip install pre-built wheels similar to upstream release flow. Otherwise, we need to maintain the HEAD-only resources. Virtualenv logic is also unavoidable unless we create linked formulae for Jinja/etc, which may cause issues if we ever hit version API compatibility problems.

Reply:
This update contains security fixes. Because of that I'd say addressing `--HEAD` should really be in a separate PR, so we can properly judge whether it is really worth the effort.